kernel/sysctl_setter.go

```diff
@@ -28,26 +28,19 @@ func NewSysctlSetter(
 }
 
 func (s *SysctlSetterImpl) Set() error {
-	values, err := Parameter(s.paramName)
-	if err != nil {
+	// always set the new value, even if the values is already set
+	// otherwise when a new USB adapter is connected, even if net.ipv6.conf.all.disable_ipv6=0 the new adaptor will have IPv6
+	// so it needs to be set again to zero, to disabled IPv6 also for the new interface
+	if err := SetParameter(s.paramName, s.desiredValue); err != nil {
 		return fmt.Errorf(
-			"retrieving the value of '%s': %w",
+			"setting the value of '%s' to %d: %w",
 			s.paramName,
+			s.desiredValue,
 			err,
 		)
 	}
-	if values[s.paramName] == s.unwantedValue {
-		err := SetParameter(s.paramName, s.desiredValue)
-		if err != nil {
-			return fmt.Errorf(
-				"setting the value of '%s' to %d: %w",
-				s.paramName,
-				s.desiredValue,
-				err,
-			)
-		}
-		s.changed = true
-	}
+	s.changed = true
```

Comment:
We still need to know whether the value was actually changed so user's configuration is not broken. Now if value was already set to a desired value by the user, it will be set to unwanted value during unset